lib/phobos/deep_struct.rb

```diff
@@ -0,0 +1,16 @@
+module Phobos
+  class DeepStruct
+    def self.extract(attrs)
+      keys = attrs.keys
+      Struct.new(*keys).new(*keys.map do |k|
```

Comment:
I was expecting the "factory" method `extract` to return a DeepStruct instance. 
So I suggest using either a module, or making it a child of Struct, so that we do have a DeepStruct instance =)
Thoughts?